File: Ser2023/Store/Store.Web/Modules/ServerTypes/Store/DocumentTypeRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface DocumentTypeRow {
    DocumentTypeId?: number;
    DocumentName?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class DocumentTypeRow {
    static readonly idProperty = 'DocumentTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'DocumentName';
    static readonly localTextPrefix = 'Store.DocumentType';
    static readonly lookupKey = 'Store.DocumentType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<DocumentTypeRow>('Store.DocumentType') }
    static async getLookupAsync() { return getLookupAsync<DocumentTypeRow>('Store.DocumentType') }

    static readonly deletePermission = 'Store:DocumentType:Delete';
    static readonly insertPermission = 'Store:DocumentType:Modify';
    static readonly readPermission = 'Store:DocumentType:View';
    static readonly updatePermission = 'Store:DocumentType:Modify';

    static readonly Fields = fieldsProxy<DocumentTypeRow>();
}

File: Ser2023/Company/Company.Web/Modules/Store/DocumentType/DocumentTypeRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;
using Company.Administration.Entities;
using System;

namespace Company.Store;

[ConnectionKey("Store"), Module("Store"), TableName("DocumentType")]
[DisplayName("Document Type"), InstanceName("Document Type")]
[ReadPermission(PermissionKeys.DocumentType.View)]
[ModifyPermission(PermissionKeys.DocumentType.Modify)]
[DeletePermission(PermissionKeys.DocumentType.Delete)]
[LookupScript("Store.DocumentType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class DocumentTypeRow : LoggingRow<DocumentTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Document Type Id"), Column("DocumentTypeID"), Identity, IdProperty]
    public int? DocumentTypeId { get => fields.DocumentTypeId[this]; set => fields.DocumentTypeId[this] = value; }

    [DisplayName("Document Name"), Size(50), NotNull, QuickSearch, NameProperty]
    public string DocumentName { get => fields.DocumentName[this]; set => fields.DocumentName[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public Int32Field DocumentTypeId;
        public StringField DocumentName;
        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}